PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/LinkMySQLToModuleDialog.java

```diff
@@ -0,0 +1,110 @@
+    private JPanel rootPanel;
+    private BasicLinkMySQLPanel basicPanel;
+
+    private MySQLNode node;
+    private Project project;
+    private Module module;
+
+    public LinkMySQLToModuleDialog(Project project, MySQLNode node, Module module) {
+        super(project);
+        this.project = project;
+        this.node = node;
+        this.module = module;
+        this.init();
+        // listener
+        this.setOkActionListener((data) -> doLink(data, project, this));
+    }
+
+    @Override
+    public AzureForm<LinkComposite<MySQLLinkConfig, ModuleLinkConfig>> getForm() {
+        return basicPanel;
+    }
+
+    @Override
+    protected String getDialogTitle() {
+        return "Link Azure Database for MySQL";
+    }
+
+    @Override
+    protected @Nullable JComponent createCenterPanel() {
+        return rootPanel;
+    }
+
+    private void createUIComponents() {
+        // TODO: place custom component creation code here
+        MySQLLinkConfig source = MySQLLinkConfig.getDefaultConfig(node);
+        ModuleLinkConfig target = ModuleLinkConfig.getDefaultConfig(module);
+        LinkComposite<MySQLLinkConfig, ModuleLinkConfig> linker = new LinkComposite<>(source, target);
+        basicPanel = new BasicLinkMySQLPanel(project, () -> linker);
+    }
+
+    private void doLink(LinkComposite<MySQLLinkConfig, ModuleLinkConfig> linker, Project project, LinkMySQLToModuleDialog dialog) {
+        final Runnable runnable = () -> {
+            final ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
+            indicator.setIndeterminate(true);
+            DefaultLoader.getIdeHelper().invokeLater(dialog::close);
+            ModulePO modulePO = createModulePO(linker.getModule());
+            MySQLServicePO mysqlPO = createMySQLPO(linker.getService());
+            LinkPO linkerPO = new LinkPO(mysqlPO.getId(), modulePO.getId(), LinkType.SERVICE_WITH_MODULE, linker.getEnvPrefix());
+            AzureMySQLStorage.getStorage().addService(mysqlPO);
+            if (ArrayUtils.isNotEmpty(linker.getService().getPasswordConfig().getPassword())) {
+                String inputPassword = String.valueOf(linker.getService().getPasswordConfig().getPassword());
+                if (MySQLConnectionUtils.connect(mysqlPO.getUrl(), mysqlPO.getUsername(), inputPassword)) {
+                    AzureMySQLStorage.getStorage().savePassword(mysqlPO, mysqlPO.getPasswordSave(), mysqlPO.getUsername(), inputPassword);
+                }
+            }
+            AzureLinkStorage.getProjectStorage(project).addLinker(linkerPO);
+        };
+        String progressMessage = "Linking Azure Database for MySQL with Module...";
+        final AzureTask task = new AzureTask(null, progressMessage, false, runnable);
+        AzureTaskManager.getInstance().runInBackground(task);
+        System.out.println("Link MySQL to Project...OK--action....");
```

Comment:
Using `Log.info` instead of `System.out.println`?

Reply:
removed